tests/unit/test_functions.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import unittest
+
+import deepdiff
+from charms.prometheus_k8s.v0.prometheus_scrape import _dedupe_job_names
+
+
```

Comment:
Do we need a test that deduping is not done in-place? I.e. that the passed list is unaffected.

Reply:
Done.